### ISSUE TITLE
Cake fails to find a member stored in an anonymous `struct` or `union`

### ISSUE THREAD

**Issue by ArthurPV** [thradams/cake#217]
Input:

```c
// a.c
struct X {
	union {
		int x;
	};
};

int main() {
	struct X x = (struct X){ .x = 50 };
}
```

Output:

```
$ ./cake -std=c2x a.c

Cake 0.9.43
/home/user/cake/src/a.c
/home/user/cake/src/a.c:8:28: error: member 'x' not found in 'X' [E720]
 8 |	struct X x = (struct X){ .x = 50 };
   |                           ~

 1 files in 0.00 seconds
 1 errors 0 warnings 0 notes
```

**Comment by thradams** [comment]
The initialization has been fixed.

```c
struct X {
	union {
		int x;
	};
};

int main() {
    constexpr struct X x = (const struct X){ .x = 50 };
    static_assert(x.x == 50);
}
```

http://thradams.com/cake/playground.html?code=Ly8gYS5jDQpzdHJ1Y3QgWCB7DQoJdW5pb24gew0KCQlpbnQgeDsNCgl9Ow0KfTsNCg0KaW50IG1haW4oKSB7DQoJY29uc3RleHByIHN0cnVjdCBYIHggPSAoY29uc3Qgc3RydWN0IFgpeyAueCA9IDUwIH07DQogICAgc3RhdGljX2Fzc2VydCh4LnggPT0gNTApOw0KfQ%3D%3D&to=-2&options=

I wish I had more unit tests with anonymous structs.

I also realized cake does not compile this code but gcc and clang compiles.

```c
struct X {
	union {
		int x;
	};
};
int main() {
    constexpr struct X x = (struct X){ .x = 50 };
    static_assert(x.x == 50);
}
```

(Cake is requiring the compound literal to be at least const)

https://godbolt.org/z/hbs5dWWzr
